lib/holdings-updater.js

```diff
@@ -0,0 +1,295 @@
+'use strict'
+
+const H = require('highland')
+let log = require('loglevel')
+const cluster = require('cluster')
+const UpdaterBase = require('./updater-base')
+
+const HoldingSierraRecord = require('./models/holding-sierra-record')
+const db = require('./db')
+const utils = require('./utils')
+const IndexerRunner = require('./indexer-runner')
+const SourceLoader = require('./sources/loader')
+const holdingSerializer = require('./serializers/holding')
+const nyplApi = require('./nyplDataApi')
+const kmsHelper = require('./kms-helper')
+
+const CACHE_TO_DISK = false
+
+class HoldingsUpdater extends UpdaterBase {
```

Comment:
This is the class that shares a lot of functionality with `BibsUpdater` and `ItemsUpdater`. I'm wondering if it makes sense to try to factor out some of those redundancies, or if that brings up too much risk of regressions. As there are three classes now the potential advantages are greater.

Reply:
That sounds like a good approach, definitely want to be careful with the code base because it seems very likely that decisions were made for reasons that I don't know or understand.